test/attr/attr_dynamic_member_lookup_default_args.swift

```diff
@@ -0,0 +1,157 @@
+// RUN: %target-run-simple-swift(-Xfrontend -disable-availability-checking)
+// REQUIRES: executable_test
+
+//===----------------------------------------------------------------------===//
+// Magic Literals
+//===----------------------------------------------------------------------===//
+
+@dynamicMemberLookup
+struct Magic: Hashable {
+  let fileID: String
+  let file: String
+  let filePath: String
+  let function: String
+  let line: Int
+  let column: Int
+  let dsohandle: UnsafeRawPointer
+
+  init(
+    fileID: String = #fileID,
```

Comment:
Do we need to test non-magic default arguments too? Probably silly and pointless to do that, but it should still not crash the compiler ;)